grpc/interceptors.go

```diff
@@ -43,3 +44,60 @@ func EnsureTimeoutUnaryClientInterceptor(
 
 	return invoker(ctx, method, req, reply, cc, opts...)
 }
+
+// The following code is for appending/extracting grpc metadata regarding module names/origins via
```

Comment:
Shoving this all inside of `grpc/interceptors.go` seemed like a good spot, but open to alternatives. At first I tried inside the robot web server or modules directory (can't remember), but hit go import cycles.

Reply:
This file seems like the appropriate place IMO.